megamek/src/megamek/server/Server.java

```diff
@@ -17783,11 +17118,11 @@ private void resolveClubAttack(PhysicalResult pr, int lastEntityId) {
             }
         }
 
-        if (((MiscType) caa.getClub().getType())
-                    .hasSubType(MiscType.S_WRECKING_BALL)
-            && (ae instanceof SupportTank) && (te instanceof Mech)) {
+        if (caa.getClub().getType().hasSubType(MiscType.S_WRECKING_BALL)
+                && (ae instanceof SupportTank) && (te instanceof Mech)) {
             // forces a PSR like a charge
             if (ae instanceof LandAirMech && ae.isAirborneVTOLorWIGE()) {
```

Comment:
My IDE suggests this is a bug, as it is always false. Is it supposed to be ||, not &&?

Reply:
It is a bug. It should be `te instanceof LandAirMech`. When an airborne LAM in AirMech mode is the target of a successful charge attack, it makes a control roll instead of a PSR. I don't know why it's restricting the attacker to a support vehicle, though. I also can't find a rule that says a successful hit is like a charge. The physical attacks table on TW, p. 214 says that on a hit roll of 2, the attacker deals damage to itself and has to make a PSR. I think this whole block may be wrong but it needs a more thorough rules check.